packages/database/src/core/util/libs/parser.ts

```diff
@@ -119,13 +138,21 @@ export const parseURL = function(
       dataURL = dataURL.substring(colonInd + 2);
     }
 
-    // Parse host and path.
+    // Parse host, path, and query string.
     let slashInd = dataURL.indexOf('/');
     if (slashInd === -1) {
       slashInd = dataURL.length;
     }
-    host = dataURL.substring(0, slashInd);
-    pathString = decodePath(dataURL.substring(slashInd));
+    let questionInd = dataURL.indexOf('?');
+    if (questionInd === -1) {
+      questionInd = dataURL.length;
+    }
+    host = dataURL.substring(0, Math.min(slashInd, questionInd));
+    if (slashInd < questionInd) {
```

Comment:
Since you are setting `questionInd` to dataUrl.length, can't you always to the substring operation below?

Reply:
no because the SDK appends a trailing `/` to everything prior to getting here so sometimes we get host = `localhost:9000?ns=foo/`. When we try to figure out the pathString for this, it'll give us the queryString instead.

```javascript
'localhost:9000?ns=foo/'.indexOf('?')
14
'localhost:9000?ns=foo/'.indexOf('/')
21
'localhost:9000?ns=foo/'.substring(21, 14);
"?ns=foo"
```